### ISSUE TITLE
Nuke line 25? : docker rust/android/build.sh: addgroup: The GID `20' is already in use.

### ISSUE THREAD

**Issue by skyl** [zingolabs/zingo-mobile#78]
Running @dev 02ee7fe88d8853b796bde135a5778fea52f83005 on x86 MBP 12.4 

```
zingo-mobile/rust/android                                                               
⚡ ./build.sh 
[+] Building 2.2s (6/6) FINISHED                                                                                                                                      
 => [internal] load build definition from Dockerfile                                                                                                             0.0s
 => => transferring dockerfile: 37B                                                                                                                              0.0s
 => [internal] load .dockerignore                                                                                                                                0.0s
 => => transferring context: 2B                                                                                                                                  0.0s
 => [internal] load metadata for docker.io/library/debian:bullseye-slim                                                                                          2.1s
 => [apt_upgrade_rustndk_image 1/2] FROM docker.io/library/debian:bullseye-slim@sha256:f6957458017ec31c4e325a76f39d6323c4c21b0e31572efa006baa927a160891          0.0s
 => CACHED [apt_upgrade_rustndk_image 2/2] RUN apt-get update  && apt-get upgrade -y  && apt-get install -y --no-install-recommends --no-install-suggests        0.0s
 => exporting to image                                                                                                                                           0.0s
 => => exporting layers                                                                                                                                          0.0s
 => => writing image sha256:c1cd51684f2e1a544b7f6028d6ac98bbcf3165016d30b51956d0a9cc99ed9536                                                                     0.0s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
[+] Building 1.1s (6/45)                                                                                                                                              
 => [internal] load build definition from Dockerfile                                                                                                             0.0s
 => => transferring dockerfile: 37B                                                                                                                              0.0s
 => [internal] load .dockerignore                                                                                                                                0.0s
 => => transferring context: 2B                                                                                                                                  0.0s
 => [internal] load metadata for docker.io/library/debian:bullseye-slim                                                                                          0.5s
 => [apt_upgrade_rustndk_image 1/2] FROM docker.io/library/debian:bullseye-slim@sha256:f6957458017ec31c4e325a76f39d6323c4c21b0e31572efa006baa927a160891          0.0s
 => CACHED [apt_upgrade_rustndk_image 2/2] RUN apt-get update  && apt-get upgrade -y  && apt-get install -y --no-install-recommends --no-install-suggests        0.0s
 => ERROR [add_builder_user 1/8] RUN addgroup --gid 20 builderuser                                                                                               0.4s
------                                                                                                                                                                
 > [add_builder_user 1/8] RUN addgroup --gid 20 builderuser:
#6 0.403 addgroup: The GID `20' is already in use.
------
executor failed running [/bin/sh -c addgroup --gid $GUID builderuser]: exit code: 1
    Finished release [optimized] target(s) in 9.43s
    Finished release [optimized] target(s) in 10.06s
    Finished release [optimized] target(s) in 8.92s
    Finished release [optimized] target(s) in 10.82s
```

**Comment by dannasessha** [comment]
Acknowledged, @skyl thank you for filing this issue.

**Comment by dannasessha** [comment]
After working with this some more, we managed to build after deleting [line 25](https://github.com/zingolabs/zingo-mobile/blob/02ee7fe88d8853b796bde135a5778fea52f83005/rust/android/Dockerfile#L25) in [zingo-mobile](https://github.com/zingolabs/zingo-mobile)/[rust](https://github.com/zingolabs/zingo-mobile/tree/dev/rust)/[android](https://github.com/zingolabs/zingo-mobile/tree/dev/rust/android)/[Dockerfile](https://github.com/zingolabs/zingo-mobile/blob/dev/rust/android/Dockerfile).

**Comment by skyl** [comment]
> After working with this some more, we managed to build after deleting [line 25](https://github.com/zingolabs/zingo-mobile/blob/02ee7fe88d8853b796bde135a5778fea52f83005/rust/android/Dockerfile#L25) in [zingo-mobile](https://github.com/zingolabs/zingo-mobile)/[rust](https://github.com/zingolabs/zingo-mobile/tree/dev/rust)/[android](https://github.com/zingolabs/zingo-mobile/tree/dev/rust/android)/[Dockerfile](https://github.com/zingolabs/zingo-mobile/blob/dev/rust/android/Dockerfile).

yep, it's still running but it's past the docker part with no problem on OSX now

**Comment by dannasessha** [comment]
I think the `debian:bullseye-slim` container is using the same kernel as your OS X. It has a value, a groupid, which is a number, for `GUID`. In OS X it likely has a user name assigned to the number. But it doesn't have a name for it yet in the container: line 25 is trying to add a group with this name (builderuser), and the `--gid` option forces the new groupid to be the given number. In your case it's 20, which is was "sniffed" from your environment by the build script and "passed in" to docker.

But, the `--gid` option is also given on the following line, saying "add this new user to the existing group"! It _puts_ the user into the group.

If we need to assign a name to the group in the container for some reason, we could find another way potentially. So now I think we should probably just nuke line 25.

**Comment by dannasessha** [comment]
I'm still not sure why this issue, so far, only showed up in OS X though.

**Comment by skyl** [comment]
> I'm still not sure why this issue, so far, only showed up in OS X though.

I always run docker without sudo. I think that might be relevant.

**Comment by dannasessha** [comment]
Running docker without `sudo` or `root` at all should be a valid, new issue. I see #14 , but this is possibly even more strict, so I'll make a new issue.

**Comment by dannasessha** [comment]
Potentially solved by #77 .

**Comment by zancas** [comment]
I believe the use of `docker cp` solves this issue.
